src/pystack/__main__.py

```diff
@@ -357,9 +357,24 @@ def process_core(parser: argparse.ArgumentParser, args: argparse.Namespace) -> N
         print(format_failureinfo_information(corefile_analyzer.extract_failure_info()))
 
     if not is_elf(executable):
-        raise errors.InvalidExecutable(
-            f"The provided executable ({executable}) doesn't have an executable format"
+        first_map = next(
```

Comment:
Should we maybe guard this with `if args.executable is not None:` to avoid falling back to the first file mapping if the user provided an executable on the command line and it's not valid?

It makes sense to me that, if our attempt to extract the executable from the core file gave us a non-ELF file, we should try again. It doesn't make sense to me for us to override the explicit choice provided by the user.

Reply:
Makes sense to me